src/Kestrel/DefaultHttpsProvider.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal;
+
+namespace Microsoft.AspNetCore.Server.Kestrel
+{
+    internal class DefaultHttpsProvider : IDefaultHttpsProvider
+    {
+        private const string AspNetHttpsOid = "1.3.6.1.4.1.311.84.1.1";
+
+        public void ConfigureHttps(ListenOptions listenOptions)
+        {
+            listenOptions.UseHttps(FindDevelopmentCertificate());
+        }
+
+        private static X509Certificate2 FindDevelopmentCertificate()
+        {
+            using (var store = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+            {
+                store.Open(OpenFlags.ReadOnly);
+
+                var certificates = store.Certificates.OfType<X509Certificate2>();
+                var certificate = certificates
```

Comment:
You need to check that it has an OID, a private key and hasn't expired

Reply:
When do you think you will have https://github.com/aspnet/Common/pull/263 merged? We shouldn't replicate all that logic here. I should be able to get the certificate by calling `CertificateManager.ListCertificates(CertificatePurpose.HTTPS, StoreName.My, StoreLocation.CurrentUser, isValid: true)`, right?

Reply:
I'm working on it